### PR TITLE
Rename DatasetTaskRef to TaskOutletDatasetReference

### PR DESCRIPTION
**Pull request by dstandish**
Makes more sense to name with `<thing doing the referencing>_<thing that is referenced>`.

### CODE REVIEW

**Review by blag — APPROVED** [review]

File: airflow/api_connexion/openapi/v1.yaml [hunk 1]
```diff
@@ -3526,10 +3526,10 @@ components:
         producing_tasks:
             type: array
             items:
-              $ref: '#/components/schemas/DatasetTaskRef'
+              $ref: '#/components/schemas/TaskDatasetProducer'
 
 
-    DatasetTaskRef:
+    TaskDatasetProducer:
```
Comment (kaxil) [inline airflow/api_connexion/openapi/v1.yaml]: wdyt about just `DatasetProducer` or `DatasetProducerTask`instead?

Comment (norm) [reply airflow/api_connexion/openapi/v1.yaml]: I like `DatasetProducerTask` the most.

Comment (potiuk) [reply airflow/api_connexion/openapi/v1.yaml]: Concur

**Review by ephraimbuddy — APPROVED** [review]

File: airflow/api_connexion/openapi/v1.yaml (hunk 1 above)
Comment (jedcunningham) [reply airflow/api_connexion/openapi/v1.yaml]: Maybe `DatasetProducingTask`? That feels a little more natural to me.

Comment (dstandish) [reply airflow/api_connexion/openapi/v1.yaml]: yeah i too am unsatisfied with this (and btw i appreciate you still taking a critical look at this even though sentiment is mostly behind the one option)

sort of by accident, i arrived at naming that feels somewhat better.

i had a reason to define relationships on DagModel in another PR, [here](https://github.com/apache/airflow/pull/25959/files#diff-62c8e300ee91e0d59f81e0ea5d30834f04db71ae74f2e155a10b51056b00b59bR3029-R3030).

And I think that sort of clarified what would be somewhat better naming.

1. rename `DatasetTaskRef` to `TaskOutletDatasetReference`
2. rename `DatasetDagRef` to `DagScheduleDatasetReference`

think about what these objects are.... are they "dataset producer tasks" or "dataset producing tasks"?  no.  they are "dataset references stored in task outlets".... so.... why not call them "task outlet dataset references".  

similar with the other one ... they are "dag schedule dataset references"....  seems pretty simple and straightforward and logical to me.  also resolves the `<from>_<to>` convention problem.

wdyt

Comment (blag) [reply airflow/api_connexion/openapi/v1.yaml]: ted-lasso-yes

Comment (ashb) [reply airflow/api_connexion/openapi/v1.yaml]: Yup, those names are much more explicit about what we use them for :+1:

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**